packages/client/hmi-client/src/components/mathml/math-editor.vue

```diff
@@ -0,0 +1,158 @@
+<template>
+	<section class="math-editor">
+		<section class="mathlive" v-if="props.mathmode === 'mathLIVE'">
+			<math-field
+				class="equation"
+				@change="updateMathLiveValue"
+				ref="mf"
+				virtual-keyboard-mode="false"
+				><slot v-if="props.mathmode === 'mathLIVE'"></slot
+			></math-field>
+		</section>
+		<section class="mathjax-container" v-if="props.mathmode === 'mathJAX'">
+			<vue-mathjax :formula="`$$${formula}$$`"></vue-mathjax>
+			<Textarea
+				v-model="jaxFormula"
+				class="mathjax"
+				id="mathjax"
+				type="text"
+				@change="updateMathJaxValue"
+				rows="2"
+				aria-label="mathjax"
+			/>
+		</section>
+	</section>
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, watch, onUpdated } from 'vue';
+import { Mathfield } from 'mathlive';
+import Mathml2latex from 'mathml-to-latex';
+import { logger } from '@/utils/logger';
+import Textarea from 'primevue/textarea';
+
+const mf = ref<Mathfield | null>(null);
+const formula = ref<string | undefined>('');
+const jaxFormula = ref<string | undefined>('');
+
+const emit = defineEmits(['formula-updated', 'mathml', 'latex']);
+
+const props = defineProps({
+	// Formula to be populated
+	value: {
+		type: String,
+		required: true
+	},
+	// Math Renderer mode
+	mathmode: {
+		type: String,
+		default: 'mathJAX'
+	}
+});
+
+const updateMathLiveValue = () => {
+	formula.value = mf.value?.getValue('latex-unstyled');
+	emit('formula-updated', formula.value);
+};
+
+const updateMathJaxValue = () => {
+	emit('formula-updated', jaxFormula.value);
+};
+
+// simple check to test if it's a mathml string
+const isMathML = (val: string): boolean => {
+	const regex = /<math[>\s][\s\S]*?<\/math>/s;
+	return regex.test(val);
+};
+
+const setPropsFormula = () => {
+	if (isMathML(props.value)) {
+		try {
+			formula.value = Mathml2latex.convert(props.value);
+		} catch (error) {
+			logger.error(error, { showToast: false, silent: true });
+		}
+	} else {
+		formula.value = props.value;
+	}
+};
+
+onUpdated(() => {
+	setPropsFormula();
+	if (props.mathmode === 'mathLIVE') {
+		mf.value?.setValue(formula.value);
+	}
+});
+
+onMounted(() => {
+	setPropsFormula();
+	if (props.mathmode === 'mathLIVE') {
+		mf.value?.setValue(formula.value);
+	}
+});
```

Comment:
Wasn't aware of this.  Not sure if I'm doing it wrong but it doesn't seem to provide the expected reactivity.

Reply:
I see, in that case it might be better to refactor the following task into it's own function to simplify `onUpdated` and `onMounted`
```js
setPropsFormula();
if (props.mathmode === 'mathLIVE') {
	mf.value?.setValue(formula.value);
}
```